lib/hub-clusters-creator/providers/schema.yaml

```diff
@@ -0,0 +1,626 @@
+          account under the Cloud IAM within the project, adding the
+          permissions 'Compute Admin' role to the service account via IAM
+          tab. Once done you can create a key under 'Service Accounts'
+          and copy and paste the JSON payload here.
+        default: ''
+        pattern: ^.*$
+
+      project:
+        $id: '#/config/project'
+        type: string
+        title: GCP Project
+        description: >
+          The name of the GCP project you are provisioning the cluster done.
+        default: ''
+        pattern: ^.*$
+
+      region:
+        $id: '#/config/region'
+        type: string
+        title: GCP Region
+        description: >
+          The GCP region you wish to the cluster to reside within.
+        default: ''
+        examples:
+          - europe-west2
+        pattern: ^.*$
+
+#
+## Cluster configuration
+#
+defaults:
+  aks:
+    defaults:
+      machine_type:
+        default: Standard_DS2_v2
+        examples:
+          - Standard_DS2_v2
+      version:
+        default: 1.14.3
+        examples:
+          - 1.14.3
+          - 1.13.0
+    required:
+      - disk_size_gb
+      - domain
+      - machine_type
+      - name
+      - services_ipv4_cidr
+      - size
+      - ssh_key
+      - version
+  gke:
+    defaults:
+      machine_type:
+        default: n1-standard-1
+        examples:
+          - n1-standard-1
+      version:
+        default: 1.13.7-gke.8
+        examples:
+          - 1.13.7-gke.8
+          - latest
+    required:
+      - description
+      - disk_size_gb
+      - domain
+      - enable_autoscaler
+      - enable_autoupgrade
+      - enable_binary_authorization
+      - enable_horizontal_pod_autoscaler
+      - enable_http_loadbalancer
+      - enable_istio
+      - enable_logging
+      - enable_monitoring
+      - enable_private_endpoint
+      - enable_private_network
+      - image_type
+      - machine_type
+      - maintenance_window
+      - name
+      - network
+      - size
+      - subnetwork
+      - version
+
+schema:
+  type: object
+  title: Cluster configuration
+  description: >
+    The configuration options for the Kubernetes cluster
+  additionalProperties: false
+
+  properties:
+    name:
+      $id: '#/properties/name'
+      type: string
+      title: Cluster Name
+      default: ''
+      description: >
+        The name of the cluster you are provision within the cloud provider
+      examples:
+        - dev
+        - prod
+      pattern: ^.*$
+
+    description:
+      $id: '#/properties/description'
+      provider: [gke]
+      type: string
+      title: Description
+      description: >
+        A summary description for this cluster.
+      default: ''
+      examples:
+        - Dev Cluster
+      pattern: ^(.*)$
+
+    domain:
+      $id: '#/properties/domain'
+      type: string
+      title: DNS Domain
+      default: ''
+      description: >
+        The dns domain which the cluster is using; this must be accessible from
```

Comment:
All DNS records for the cluster will created within this domain (zone) which must be accessible from the cloud project.